iib/workers/tasks/opm_operations.py

```diff
@@ -744,3 +751,133 @@ def opm_create_empty_fbc(
+        # migated_catalog_dir path will look like /tmp/iib-**/catalog
+        migated_catalog_dir, _ = opm_migrate(
+            index_db=index_db_path,
+            base_dir=temp_dir,
+            generate_cache=False,
+        )
+        log.info("Migated catalog after removing from db at %s", migated_catalog_dir)
+
+        # copy the content of migrated_catalog to from_index's config
+        log.info("Copying content of %s to %s", migated_catalog_dir, from_index_configs_dir)
+        for operator_package in os.listdir(migated_catalog_dir):
+            shutil.copytree(
+                os.path.join(migated_catalog_dir, operator_package),
+                os.path.join(from_index_configs_dir, operator_package),
+                dirs_exist_ok=True,
+            )
+
+    # copy fragment_operator to from_index configs
+    fragment_opr_dest_path = os.path.join(from_index_configs_dir, fragment_operator)
+    if os.path.exists(fragment_opr_dest_path):
+        shutil.rmtree(fragment_opr_dest_path)
+    log.info(
+        "Copying content of %s to %s",
+        os.path.join(fragment_path, fragment_operator),
+        fragment_opr_dest_path,
+    )
+    shutil.copytree(os.path.join(fragment_path, fragment_operator), fragment_opr_dest_path)
+
+    local_cache_path = os.path.join(temp_dir, 'cache')
+    generate_cache_locally(
+        base_dir=temp_dir, fbc_dir=from_index_configs_dir, local_cache_path=local_cache_path
+    )
+
+    log.info("Dockerfile generated from %s", from_index_configs_dir)
+    opm_generate_dockerfile(
+        fbc_dir=from_index_configs_dir,
+        base_dir=temp_dir,
+        index_db=index_db_path,
+        binary_image=binary_image,
+        dockerfile_name='index.Dockerfile',
+    )
+
+
+def verify_operator_exists(
+    from_index: str,
+    base_dir: str,
+    operator_package: str,
+    overwrite_from_index_token: Optional[str],
+):
+    """
+    Check if operator exists in index image.
+
+    :pararm str from_index: index in which operator existence is checke
+    :param str base_dir: base temp directory for IIB request
+    :pararm str operator_package: operator_package to check
+    :pararm str overwrite_from_index_token: token used to access the image
+    :return is_package_in_index, index_db_path
+    :rtype: (str, str)
+    """
+    from iib.workers.tasks.build import terminate_process, get_bundle_json
+    from iib.workers.tasks.iib_static_types import BundleImage
+    from iib.workers.tasks.utils import run_cmd
+    from iib.workers.tasks.utils import set_registry_token
+
+    is_package_in_index = False
+
+    log.info("Verifying if operator package %s exists in index %s", operator_package, from_index)
+
+    # check if operater package exists in hidden index.db
+    with set_registry_token(overwrite_from_index_token, from_index, append=True):
+        index_db_path = get_hidden_index_database(from_index=from_index, base_dir=base_dir)
+
+    port, rpc_proc = opm_registry_serve(db_path=index_db_path)
```

Comment:
Lines 871 to 876 can be replaced with a call to `https://github.com/release-engineering/iib/blob/master/iib/workers/tasks/utils.py#L1212`

Reply:
so if I call grpcurl_get_db_data() here, it will call opm_serve_from_index() which will end up serving data from catalog dir(as we will have from_index as fbc), but we want to specifically call opm_registry_serve() to serve from db to check the existence of operator package in db, hence I had to duplicate some code.
Please let me know if I missed something

Reply:
So, if I check the code here: https://github.com/release-engineering/iib/blob/master/iib/workers/tasks/opm_operations.py#L49, it will indeed call opm_registry_serve for FBC indexes. Am I missing something?

Reply:
oh shoot! I'm sorry. you're right